stun-types/src/lib.rs

```diff
@@ -98,7 +98,8 @@ impl FromStr for TransportType {
     type Err = ParseTransportTypeError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
+        let value = alloc::string::String::from(s).to_uppercase();
+        match value.as_str() {
```

Comment:
```suggestion
        match value.to_ascii_uppercase().as_str() {
```

Saves one allocation.

Or if we want to save two allocations, use an if ladder with `str::eq_ignore_ascii_case()`

Reply:
For references, this requires not storing value altogether. Now there are 3 allocations :)